clients/venice-client/src/main/java/com/linkedin/venice/fastclient/InternalAvroStoreClient.java

```diff
@@ -1,58 +1,207 @@
 package com.linkedin.venice.fastclient;
 
 import com.linkedin.venice.client.exceptions.VeniceClientException;
-import com.linkedin.venice.client.store.AvroGenericStoreClient;
-import com.linkedin.venice.client.store.ComputeRequestBuilder;
+import com.linkedin.venice.client.store.AvroGenericReadComputeStoreClient;
+import com.linkedin.venice.client.store.ComputeGenericRecord;
 import com.linkedin.venice.client.store.streaming.StreamingCallback;
+import com.linkedin.venice.client.store.streaming.VeniceResponseCompletableFuture;
 import com.linkedin.venice.client.store.streaming.VeniceResponseMap;
+import com.linkedin.venice.client.store.streaming.VeniceResponseMapImpl;
+import com.linkedin.venice.compute.ComputeRequestWrapper;
+import com.linkedin.venice.read.RequestType;
+import com.linkedin.venice.router.exception.VeniceKeyCountLimitException;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
+import java.util.Queue;
 import java.util.Set;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
 
 
 /**
  * All the internal implementations of different tiers should extend this class.
  * This class adds in {@link RequestContext} object for the communication among different tiers.
  */
 
-public abstract class InternalAvroStoreClient<K, V> implements AvroGenericStoreClient<K, V> {
+public abstract class InternalAvroStoreClient<K, V> implements AvroGenericReadComputeStoreClient<K, V> {
+  public abstract ClientConfig getClientConfig();
+
+  @Override
+  public final boolean isProjectionFieldValidationEnabled() {
+    return getClientConfig().isProjectionFieldValidationEnabled();
+  }
+
+  @Override
   public CompletableFuture<V> get(K key) throws VeniceClientException {
     return get(new GetRequestContext(), key);
   }
 
   protected abstract CompletableFuture<V> get(GetRequestContext requestContext, K key) throws VeniceClientException;
 
+  @Override
+  public final CompletableFuture<Map<K, V>> batchGet(Set<K> keys) throws VeniceClientException {
+    BatchGetRequestContext<K, V> batchGetRequestContext = new BatchGetRequestContext<>();
+    // Since user has invoked batchGet directly, then we do not want to allow partial success
+    batchGetRequestContext.isPartialSuccessAllowed = false;
+    return batchGet(batchGetRequestContext, keys);
+  }
+
+  protected CompletableFuture<Map<K, V>> batchGet(BatchGetRequestContext<K, V> requestContext, Set<K> keys)
+      throws VeniceClientException {
+    // Since user has invoked batchGet directly, then we do not want to allow partial success
+    requestContext.isPartialSuccessAllowed = false;
```

Comment:
Changing the constructor is a breaking change to the API and I want to avoid it. It is also not necessary as the name is sufficient: `batchGet` - don't allow partial response, `streamingBatchGet` - allow partial responses.

The reason I added it here is that although this API is internal only, in tests we call it directly and it would miss setting this field

I can remove it from the API without the `BatchGetRequestContext`, but left it there to be a bit pedantic. Let me know if you think it should be removed from the `batchGet(Set<K> keys)` API

Reply:
Oh, I now get what you mean. You meant constructor of `BatchGetRequestContext`, right? Hmm. That's a good idea. I will try it out

Reply:
yes, I meant the constructor of `BatchGetRequestContext`